omnigibson/systems/system_base.py

```diff
@@ -1,50 +1,40 @@
 import omnigibson as og
 from omnigibson.macros import gm
-from omnigibson.utils.python_utils import classproperty, assert_valid_key, \
+from omnigibson.utils.python_utils import classproperty, assert_valid_key, get_uuid, camel_case_to_snake_case, \
     SerializableNonInstance, UniquelyNamedNonInstance
 from omnigibson.utils.registry_utils import SerializableRegistry
 from omnigibson.utils.ui_utils import create_module_logger
 
 # Create module logger
 log = create_module_logger(module_name=__name__)
 
-# Global dicts that will contain mappings of all the systems
-REGISTERED_SYSTEMS = dict()
-
-
-def get_system(system_name):
-    system = REGISTERED_SYSTEMS[system_name]
-    # If the scene has already been loaded when get_system is called, add the system to the registry and initialize
-    if og.sim and og.sim.scene and og.sim.scene.loaded and og.sim.scene.system_registry("name", system_name) is None:
-        og.sim.scene.system_registry.add(obj=system)
-        system.initialize()
-    return system
-
-
 class BaseSystem(SerializableNonInstance, UniquelyNamedNonInstance):
     """
     Base class for all systems. These are non-instance objects that should be used globally for a given environment.
     This is useful for items in a scene that are non-discrete / cannot be distinguished into individual instances,
     e.g.: water, particles, etc.
     """
     def __init_subclass__(cls, **kwargs):
+        # While class names are camel case, we convert them to snake case to be consistent with object categories.
+        cls._snake_case_name = camel_case_to_snake_case(cls.__name__)
```

Comment:
not a huge fan of this, but i guess this is necessary in order to maintain both snake case string reference and camel case class naming?

Reply:
Explain why we want the snake case for systems.

Reply:
done